### PR TITLE
Added Enhancement in the app to let user delete single note as well r…

### PR DESCRIPTION
**Pull request by shabirjan**
…ather than emptying whole trash bin.

### CODE REVIEW

**Comment by shabirjan** [comment]
@stephdau @markryall @folletto @frosty @koke any response on my pull request?

**Comment by folletto** [comment]
I'm not very aware of the design patterns for Simplenote, but in general — could you add a screenshot or better a gif/video so it's quickly visible the change you're suggesting?

**Comment by shabirjan** [comment]
This is the enhancement i have worked on as per : #143 
[GIF](https://www.dropbox.com/s/olahdkjjbrizcex/out.gif?dl=0)

**Comment by folletto** [comment]
Design wise, looks like a good enhancement. 

Two considerations:

1. Label: it shouldn't be "Trash" for both moving to trash and deleting permanently. Maybe just "Delete" would do. Other ideas?
2. Order: any thought about the order? Given the kind of action, I'd expect a user most often trying to "Restore", so I'd put that right-most.

**Comment by shabirjan** [comment]
1. Delete will be fine
2. Agree

**Comment by shabirjan** [comment]
@folletto So should i improve these and commit the changes?

**Comment by folletto** [comment]
I'm not sure in terms of merging and code review, but design wise yes, I think these two changes should happen.

**Comment by koke** [comment]
I think @jleandroperez is the best one to review the code

**Comment by shabirjan** [comment]
@folletto both changes are done.

File: SimplenoteShare/Base.lproj/MainInterface.storyboard [hunk 1]
```diff
@@ -1,21 +1,25 @@
-<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (jleandroperez) [inline SimplenoteShare/Base.lproj/MainInterface.storyboard]: Can we revert the changes on this file? (since it's not really needed for this PR!)

File: Simplenote/Classes/SPNoteListViewController.m [hunk 2]
```diff
@@ -511,28 +505,42 @@ - (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)
...
-        if (note.deleted) {
-            [[SPObjectManager sharedManager] restoreNote:note];
-            [[CSSearchableIndex defaultSearchableIndex] indexSearchableNote:note];
-        } else {
-            [SPTracker trackListNoteDeleted];
-            [[SPObjectManager sharedManager] trashNote:note];
-            [[CSSearchableIndex defaultSearchableIndex] deleteSearchableNote:note];
-        }
+        return  @[restore,permanentDelete];
+    }else{
```
Comment (jleandroperez) [inline Simplenote/Classes/SPNoteListViewController.m]: Missing spacing `} else {`

File: Simplenote/Classes/SPNoteListViewController.m [hunk 3]
```diff
@@ -511,28 +505,42 @@ - (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)
...
+- (NSArray *)tableView:(UITableView*)tableView editActionsForRowAtIndexPath:(nonnull NSIndexPath *)indexPath{
     
-    return UITableViewCellEditingStyleDelete;
-}
-
-- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
+    Note *note = [self.fetchedResultsController objectAtIndexPath:indexPath];
     
-    if (editingStyle == UITableViewCellEditingStyleDelete) {
+    UITableViewRowAction *delete = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleDefault title:NSLocalizedString(@"Trash-verb", @"Trash (verb) - the action of deleting a note") handler:^(UITableViewRowAction *action, NSIndexPath *indexPath)
```
Comment (jleandroperez) [inline Simplenote/Classes/SPNoteListViewController.m]: Maybe rename this var to trash?

File: Simplenote/Classes/SPNoteListViewController.m [hunk 4]
```diff
@@ -511,28 +505,42 @@ - (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)
...
+    delete.backgroundColor = [UIColor redColor];
+    
+    UITableViewRowAction *restore = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleDefault title:NSLocalizedString(@"Restore", @"Restore a note from the trash, markking it as undeleted")  handler:^(UITableViewRowAction *action, NSIndexPath *indexPath)
+                                    {
+                                        [[SPObjectManager sharedManager] restoreNote:note];
+                                        [[CSSearchableIndex defaultSearchableIndex] indexSearchableNote:note];
+                                    }];
+    restore.backgroundColor = [UIColor blueColor];
+   
+    UITableViewRowAction *permanentDelete = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleDefault title:NSLocalizedString(@"Delete", @"Trash (verb) - the action of deleting a note") handler:^(UITableViewRowAction *action, NSIndexPath *indexPath)
```
Comment (jleandroperez) [inline Simplenote/Classes/SPNoteListViewController.m]: (And maybe rename this one to `delete`?)

File: Simplenote/Classes/SPNoteListViewController.m [hunk 5]
```diff
@@ -511,28 +505,42 @@ - (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)
...
+                                        [[CSSearchableIndex defaultSearchableIndex] deleteSearchableNote:note];
+                                    }];
+    delete.backgroundColor = [UIColor redColor];
+    
+    UITableViewRowAction *restore = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleDefault title:NSLocalizedString(@"Restore", @"Restore a note from the trash, markking it as undeleted")  handler:^(UITableViewRowAction *action, NSIndexPath *indexPath)
+                                    {
+                                        [[SPObjectManager sharedManager] restoreNote:note];
+                                        [[CSSearchableIndex defaultSearchableIndex] indexSearchableNote:note];
+                                    }];
+    restore.backgroundColor = [UIColor blueColor];
```
Comment (jleandroperez) [inline Simplenote/Classes/SPNoteListViewController.m]: Maybe change this one to `orangeColor`? (Restore action in blue, while the permanently delete is in read, looks honestly a bit odd!). Just borrowing color ideas from Apple Mail.app, which displays actions in Gray / Orange / Red.

File: Simplenote/Classes/SPNoteListViewController.m [hunk 6]
```diff
@@ -511,28 +505,42 @@ - (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)
...
         
-        if (note.deleted) {
-            [[SPObjectManager sharedManager] restoreNote:note];
-            [[CSSearchableIndex defaultSearchableIndex] indexSearchableNote:note];
-        } else {
-            [SPTracker trackListNoteDeleted];
-            [[SPObjectManager sharedManager] trashNote:note];
-            [[CSSearchableIndex defaultSearchableIndex] deleteSearchableNote:note];
-        }
+        return  @[restore,permanentDelete];
```
Comment (jleandroperez) [inline Simplenote/Classes/SPNoteListViewController.m]: Order should (probably) be inverted. When the user swipes over the cell, the actual actions order is **Delete / Restore**, inverting the buttons here should result in the **Delete** action to the right.

File: Simplenote/Classes/SPNoteListViewController.m [hunk 7]
```diff
@@ -511,28 +505,42 @@ - (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)
...
+                                        [[CSSearchableIndex defaultSearchableIndex] indexSearchableNote:note];
+                                    }];
+    restore.backgroundColor = [UIColor blueColor];
+   
+    UITableViewRowAction *permanentDelete = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleDefault title:NSLocalizedString(@"Delete", @"Trash (verb) - the action of deleting a note") handler:^(UITableViewRowAction *action, NSIndexPath *indexPath)
+                                    {
+                                        // Delete something here
+                                        [SPTracker trackListNoteDeleted];
+                                        [[SPObjectManager sharedManager] permenentlyDeleteNote:note];
+
```
Comment (jleandroperez) [inline Simplenote/Classes/SPNoteListViewController.m]: Nipticky: Extra newline here!

File: SimplenoteShare/Base.lproj/MainInterface.storyboard (hunk 1 above)
Comment (shabirjan) [inline SimplenoteShare/Base.lproj/MainInterface.storyboard]: Dont know whether i can do that?

File: Simplenote/Classes/SPNoteListViewController.m (hunk 3 above)
Comment (shabirjan) [inline Simplenote/Classes/SPNoteListViewController.m]: Done :)

File: Simplenote/Classes/SPNoteListViewController.m (hunk 5 above)
Comment (shabirjan) [inline Simplenote/Classes/SPNoteListViewController.m]: Done :)

File: Simplenote/Classes/SPNoteListViewController.m (hunk 4 above)
Comment (shabirjan) [inline Simplenote/Classes/SPNoteListViewController.m]: Done :)

File: Simplenote/Classes/SPNoteListViewController.m (hunk 7 above)
Comment (shabirjan) [inline Simplenote/Classes/SPNoteListViewController.m]: Removed :)

File: Simplenote/Classes/SPNoteListViewController.m (hunk 6 above)
Comment (shabirjan) [inline Simplenote/Classes/SPNoteListViewController.m]: Done :) the actual order was Restore-Delete but @folletto suggested we should change the order to Delete-Restore so i changed that but now the order is Restore-Delete

File: Simplenote/Classes/SPNoteListViewController.m (hunk 2 above)
Comment (shabirjan) [inline Simplenote/Classes/SPNoteListViewController.m]: Done :)

File: Simplenote/Classes/SPNoteListViewController.m [hunk 8]
```diff
@@ -511,28 +505,41 @@ - (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)
...
+- (NSArray *)tableView:(UITableView*)tableView editActionsForRowAtIndexPath:(nonnull NSIndexPath *)indexPath{
     
-    return UITableViewCellEditingStyleDelete;
-}
-
-- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
+    Note *note = [self.fetchedResultsController objectAtIndexPath:indexPath];
     
-    if (editingStyle == UITableViewCellEditingStyleDelete) {
+    UITableViewRowAction *trash = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleDefault title:NSLocalizedString(@"Trash-verb", @"Trash (verb) - the action of deleting a note") handler:^(UITableViewRowAction *action, NSIndexPath *indexPath)
```
Comment (frosty) [inline Simplenote/Classes/SPNoteListViewController.m]: Apologies for the drive-by comment, but I think we should probably be using `UITableViewRowActionStyleDestructive` for the deletion actions.

**Comment by markryall** [comment]
Hey @shabirjan!

I appreciate being included in this conversation with such esteemed company.

Although an avid simplenote-ios user, I am not contributor to this project however so am unable to offer much insight.

Excellent work on the PR!  I look forward to using your feature.

File: Simplenote/Classes/SPNoteListViewController.m (hunk 8 above)
Comment (shabirjan) [inline Simplenote/Classes/SPNoteListViewController.m]: Done :)

File: Simplenote/Classes/SPNoteListViewController.m (hunk 6 above)
Comment (folletto) [inline Simplenote/Classes/SPNoteListViewController.m]: As a clarification for future reference: Jorge with "actual" is referring to Apple Mail again, which is a good perspective to take here too as above. 

👍  to the change.

**Comment by jleandroperez** [comment]
@shabirjan thanks *A LOT* for contributing to Simplenote sir!!

H/T @folletto and @frosty for jumping over for the review.

:shipit:
